custom_components/localtuya/cover.py

```diff
@@ -180,108 +201,85 @@ def __init__(
-        switchid,
+        coverid,
         **kwargs,
     ):
-        #_LOGGER.info("running def __init__ of LocaltuyaCover(CoverEntity) with self=%s device=%s name=%s friendly_name=%s icon=%s switchid=%s open_cmd=%s close_cmd=%s stop_cmd=%s", self, device, name, friendly_name, icon, switchid, open_cmd, close_cmd, stop_cmd)
-        super().__init__(device, config_entry, switchid, **kwargs)
-        self._state = None
-        self._position = 50
+        super().__init__(device, config_entry, coverid, **kwargs)
+        self._position = None
+        self._current_cover_position = None
+        self._last_movement = None
+        self._last_position_set = None
+        self._last_command = None
         print(
-            "Initialized tuya cover  [{}] with switch status [{}] and state [{}]".format(
-                self.name, self._status, self._state
+            "Initialized tuya cover  [{}] with switch status [{}]".format(
+                self.name, self._status
             )
         )
 
     @property
-    def available(self):
-        """Return if device is available or not."""
-        return self._available
+    def is_closed(self):
+        """Check if the cover is fully closed."""
+        if self._current_cover_position == 100:
+            return True
+        return False
 
-    @property
-    def supported_features(self):
-        """Flag supported features."""
-        supported_features = (
-            SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_SET_POSITION
-        )
-        return supported_features
+    def status_updated(self):
+        """Device status was updated."""
+        _LOGGER.debug("status_updated called")
+        self._cached_status = self._status
+        _LOGGER.debug("status_updated called, self._status=%s, self._cached_status=%s", self._status, self._cached_status)
+        try:
+            self._last_movement = self._cached_status['dps'][str(self._config[CONF_LAST_MOVEMENT])]
+            self._last_position_set = self._cached_status['dps'][str(self._config[CONF_SET_POSITION])]
+            self._current_cover_position = self._cached_status['dps'][str(self._config[CONF_GET_POSITION])]
+            self._last_command = self._cached_status['dps'][str(self._dps_id)]
+        except:
+            _LOGGER.warning("status_updated failed to update certain variables")
 
     @property
     def current_cover_position(self):
-        # self.update()
-        # state = self._state
-        #        _LOGGER.info("curr_pos() : %i", self._position)
-        # print('curr_pos() : state [{}]'.format(state))
-        return self._position
+        """Return position of Tuya cover."""
+        return self._current_cover_position 
 
     @property
-    def is_opening(self):
-        # self.update()
-        state = self._state
-        # print('is_opening() : state [{}]'.format(state))
-        if state == "on":
-            return True
-        return False
+    def min_position(self):
```

Comment:
Both `min_position` and `max_position` are not used here and not present in the base class (`CoverEntity`), so you can remove them. The constants can be removed as well.

Reply:
Agreed.